### PR TITLE
docs: Add documentation for Gateway API Addresses Support

### PR DESCRIPTION
**Pull request by chaunceyjiang**
Part of #33042

### CODE REVIEW

File: Documentation/network/servicemesh/gateway-api/addresses.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+.. code-block:: shell-session
+
+    $ kubectl get gateway my-gateway
+    NAME         CLASS    ADDRESS        PROGRAMMED   AGE
+    my-gateway   cilium   172.18.0.141   True         2d7h
+
+
+.. note::
+
+    At a future date this ``io.cilium/lb-ipam-ips`` annotation will be removed if no ``spec.addresses`` are set.
```
Comment (ghost) [inline Documentation/network/servicemesh/gateway-api/addresses.rst]: I don't understand this note - you mean that support for the annotation will be removed from Cilium, or that Cilium will automatically remove the annotation? Does the "future date" refer to some future release?

**Review by ghost — COMMENTED** [review]
Thanks! Looks good overall, I left a question about a note I didn't understand.

File: Documentation/network/servicemesh/gateway-api/addresses.rst (hunk 1 above)
Comment (chaunceyjiang) [reply Documentation/network/servicemesh/gateway-api/addresses.rst]: > I don't understand this note - you mean that support for the annotation will be removed from Cilium, or that Cilium will automatically remove the annotation?

At this point, there's no definitive conclusion - my understanding is that this annotation may not be supported in the future, and users will be prompted to use `spec.addresses` instead.

Refer to : https://github.com/cilium/cilium/pull/33042#discussion_r1713206663

File: Documentation/network/servicemesh/gateway-api/addresses.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+.. code-block:: shell-session
+
+    $ kubectl get gateway my-gateway
+    NAME         CLASS    ADDRESS        PROGRAMMED   AGE
+    my-gateway   cilium   172.18.0.140   True         2d7h
+
+
+
+If already use the ``io.cilium/lb-ipam-ips`` in the ``spec.infrastructure.annotations`` to specify the IP, the ``spec.addresses`` field will be ignored.
```
Comment (youngnick) [inline Documentation/network/servicemesh/gateway-api/addresses.rst]:
```suggestion
If you are already using the ``io.cilium/lb-ipam-ips`` in the ``spec.infrastructure.annotations`` to specify the IP, the ``spec.addresses`` field will be ignored.
```

File: Documentation/network/servicemesh/gateway-api/addresses.rst (hunk 1 above)
Comment (youngnick) [reply Documentation/network/servicemesh/gateway-api/addresses.rst]:
```suggestion
    At a future date the use of the ``io.cilium/lb-ipam-ips`` will be deprecated, and then after that, this annotation will be ignored if no ``spec.addresses`` are set. In both cases, warning logs will be added to the Cilium agent logs, and a warning Condition will be placed on the Gateway.
```

Something like this is more what I meant in the original comment.

**Review by ghost — COMMENTED** [review]
After Nick's comments are addressed looks good. @chaunceyjiang Can you address these comments?

File: Documentation/network/servicemesh/gateway-api/addresses.rst (hunk 1 above)
Comment (chaunceyjiang) [reply Documentation/network/servicemesh/gateway-api/addresses.rst]: Done.

File: Documentation/network/servicemesh/gateway-api/addresses.rst (hunk 2 above)
Comment (chaunceyjiang) [reply Documentation/network/servicemesh/gateway-api/addresses.rst]: Done.

**Comment by chaunceyjiang** [comment]
@lambdanis PTAL.

**Review by ghost — APPROVED** [review]
Thank you @chaunceyjiang

**Pull request merged**